test/integration/evictions/evictions_test.go

```diff
@@ -216,14 +221,16 @@ func TestConcurrentEvictionRequests(t *testing.T) {
 		_, err := clientSet.Core().Pods(ns.Name).Get(podName, metav1.GetOptions{})
 		if !errors.IsNotFound(err) {
 			t.Errorf("Pod %q is expected to be evicted", podName)
+			if err == nil {
+				clientSet.Core().Pods(ns.Name).Delete(podName, deleteOption)
+			}
 		}
 	}
 
 	if err := clientSet.Policy().PodDisruptionBudgets(ns.Name).Delete(pdb.Name, deleteOption); err != nil {
 		t.Errorf("Failed to delete PodDisruptionBudget: %v", err)
```

Comment:
is there any reason to not call `t.Fatalf` here or are you thinking you'd like to be able to move this block around in the future?